### DISCUSSION TITLE
Choose between a latent and image upscaler

### DISCUSSION THREAD

**Discussion by RadioVs** [Acly/krita-ai-diffusion#400] [Ideas]
On SDXL, at resolutions up to 1536, a latent upscaler is used, this adds detail to natural shapes but breaks all the straight diagonal lines. It would be nice to be able to choose between latent and image upscaler in the generation panel, and also turn off upscaler for generation in full resolution.

Initial image 1024x1024
![ComfyUI_temp_voaxd_00020_](https://github.com/Acly/krita-ai-diffusion/assets/62827539/cdf2a2a6-1cd5-4118-b8f4-789cfc08bca0)

Result image 1480x1480 with latent upscaler
![ComfyUI_temp_bkjev_00017_](https://github.com/Acly/krita-ai-diffusion/assets/62827539/960cd42c-14a0-4dd2-8ef1-b25745eaecc5)

Result image 1480x1480 with image upscaler
![ComfyUI_temp_gpknr_00017_](https://github.com/Acly/krita-ai-diffusion/assets/62827539/af748656-74e1-4b16-9d6a-e1968c83aa65)

**Comment by ArshCypherZ** [comment]
image [upscaler](https://enhanceai.art/upscaling)

**Comment by Acly** [comment]
I agree the wonky lines are an issue with latent upscale, even at low scaling factor. I chose it only because it's faster and quality doesnt suffer much in a lot of cases, but clean line art is rather important use case.

I'm not so sure about making it a choice, the effects of latent upscale you describe are more coincidental and if someone really wanted to they could achieve similar result more deliberately.

Likewise running text-to-image at resolutions that diverge too much from the checkpoint resolution is almost always really bad and not worth an option IMO. If you really want to, you can enforce it by modifying the checkpoint resolution in Style's advanced checkpoint settings.

**Reply by RadioVs** [reply]
I got the idea of checkpoint resolution, although it is not too obvious, I think it’s worth at least adding it to the documentation, but something needs to be done with broken lines.

**Reply by Acly** [reply]
Going to remove the latent upscale entirely, I already had to remove it from inpainting workflow anyway due to crop/scaling resolution issues

**Reply by RadioVs** [reply]
Thats a great news, thank you!
